packages/palette/src/elements/Toggle/Toggle.tsx

```diff
@@ -28,91 +27,76 @@ export interface ToggleState {
 }
 
 /** A toggle component used to show / hide / expand content  */
-export class Toggle extends React.Component<ToggleProps> {
-  static defaultProps = {
-    textSize: "2",
-    chevronSize: 12,
-  }
-
-  state = {
-    expanded: false,
-    disabled: false,
-  }
-
-  constructor(props) {
-    super(props)
-
-    this.state = {
-      ...props,
-    }
-  }
+export const Toggle: React.FC<ToggleProps> = ({
+  label,
+  textSize = "2",
+  chevronSize = 12,
+  expanded: defaultExpanded,
+  disabled,
+  children,
+  renderSecondaryAction,
+  ...rest
+}) => {
+  const [expanded, setExpanded] = useState(defaultExpanded)
 
-  toggleExpand = () => {
-    if (!this.props.disabled) {
-      this.setState({
-        expanded: !this.state.expanded,
-      })
+  const toggleExpand = () => {
+    if (!disabled) {
+      setExpanded((prevExpanded) => !prevExpanded)
     }
   }
 
-  render() {
-    const { disabled, expanded } = this.state
-    const {
-      children,
-      chevronSize,
-      label,
-      textSize,
-      renderSecondaryAction,
-    } = this.props
+  return (
+    <Flex width="100%" flexDirection="column" pb={2} {...rest}>
+      <Header
+        onClick={toggleExpand}
+        disabled={disabled}
+        borderTop="1px solid"
+        borderColor="black10"
+        pt={2}
+        aria-expanded={expanded}
+      >
+        <Flex justifyContent="space-between" alignItems="center">
+          {typeof label === "string" ? (
+            <Sans
+              size={textSize as SansSize}
+              weight="medium"
+              color="black100"
+              my={0.5}
+            >
+              {label}
+            </Sans>
```

Comment:
It's weird to me that changing this to `<Text>` doesn't work because of types? This is the only place we're still using `Sans` right?

Reply:
When we change it have to also change the call sites where there would be type errors because it'd go from `textSize?: SansSize` to `variant?: TextVariant`. There may not be many at all of these in the wild; but it'd still be technically a breaking change. There's a bunch of breaking changes that I think might make sense to just handle in one go though.